sds/src/secondary_validation/dutch_passport_checksum.rs

```diff
@@ -0,0 +1,101 @@
+use crate::secondary_validation::Validator;
+
+pub struct DutchPassportChecksum;
+
+const DUTCH_PASSPORT_LENGTH: usize = 9;
+
+impl Validator for DutchPassportChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        /*
+         * Dutch passport numbers are 9 characters long:
+         * - First two characters are letters from A-N or P-Z (excluding O)
+         * - Next 6 characters can be either digits or letters from A-N or P-Z (excluding O)
+         * - Last character is a checksum digit calculated using the ICAO 9303 algorithm
+         */
+        if regex_match.len() != DUTCH_PASSPORT_LENGTH {
+            return false;
+        }
+
+        // Check first two characters are valid letters (A-N or P-Z)
+        if !regex_match[..2].chars().all(|c| {
```

Comment:
Direct indexing of a string is a bit dangerous unless you have checked that the string contains only ASCII chars, since some Unicode characters can be multiple bytes and if you index in the middle of a character you will get a panic in Rust. The regex this checksum is used with _may_ be enforcing this already, but it's best to avoid this. In this case it's easy to just use the `.chars()` iterator, in addition to `.take()` to take a certain number of chars.